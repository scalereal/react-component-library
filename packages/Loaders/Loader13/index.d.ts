import * as React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
declare const Loader13: React.FC<Props> & WithStyle;
export default Loader13;
