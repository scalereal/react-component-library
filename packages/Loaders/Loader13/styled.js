var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
import styled from 'styled-components';
export var Loader = styled.div(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    color: ", ";\n    vertical-align: middle;\n    pointer-events: none;\n    \n    transform: rotateZ(45deg);\n\tperspective: 1000px;\n\tborder-radius: 50%;\n\t&:before,\n\t&:after {\n\t\tcontent: '';\n\t\tdisplay: block;\n\t\tposition: absolute;\n\t\ttop: 0;\n\t\tleft: 0;\n\t\twidth: inherit;\n\t\theight: inherit;\n\t\tborder-radius: 50%;\n\t\tanimation: 1s spin linear infinite;\n\t}\n\t&:before {\n\t\ttransform: rotateX(70deg);\n\t}\n\t&:after {\n\t\ttransform: rotateY(70deg);\n\t\tanimation-delay: .4s;\n\t}\n\n    @keyframes rotate {\n        0% {\n            transform: translate(-50%, -50%) rotateZ(0deg);\n        }\n        100% {\n            transform: translate(-50%, -50%) rotateZ(360deg);\n        }\n    }\n\n    @keyframes rotateccw {\n        0% {\n            transform: translate(-50%, -50%) rotate(0deg);\n        }\n        100% {\n            transform: translate(-50%, -50%) rotate(-360deg);\n        }\n    }\n\n    @keyframes spin {\n        0%,\n        100% {\n            box-shadow: .2em 0px 0 0px currentcolor;\n        }\n        12% {\n            box-shadow: .2em .2em 0 0 currentcolor;\n        }\n        25% {\n            box-shadow: 0 .2em 0 0px currentcolor;\n        }\n        37% {\n            box-shadow: -.2em .2em 0 0 currentcolor;\n        }\n        50% {\n            box-shadow: -.2em 0 0 0 currentcolor;\n        }\n        62% {\n            box-shadow: -.2em -.2em 0 0 currentcolor;\n        }\n        75% {\n            box-shadow: 0px -.2em 0 0 currentcolor;\n        }\n        87% {\n            box-shadow: .2em -.2em 0 0 currentcolor;\n        }\n    }\n"], ["\n    display: inline-block;\n    width: 1em;\n    height: 1em;\n    color: ", ";\n    vertical-align: middle;\n    pointer-events: none;\n    \n    transform: rotateZ(45deg);\n\tperspective: 1000px;\n\tborder-radius: 50%;\n\t&:before,\n\t&:after {\n\t\tcontent: '';\n\t\tdisplay: block;\n\t\tposition: absolute;\n\t\ttop: 0;\n\t\tleft: 0;\n\t\twidth: inherit;\n\t\theight: inherit;\n\t\tborder-radius: 50%;\n\t\tanimation: 1s spin linear infinite;\n\t}\n\t&:before {\n\t\ttransform: rotateX(70deg);\n\t}\n\t&:after {\n\t\ttransform: rotateY(70deg);\n\t\tanimation-delay: .4s;\n\t}\n\n    @keyframes rotate {\n        0% {\n            transform: translate(-50%, -50%) rotateZ(0deg);\n        }\n        100% {\n            transform: translate(-50%, -50%) rotateZ(360deg);\n        }\n    }\n\n    @keyframes rotateccw {\n        0% {\n            transform: translate(-50%, -50%) rotate(0deg);\n        }\n        100% {\n            transform: translate(-50%, -50%) rotate(-360deg);\n        }\n    }\n\n    @keyframes spin {\n        0%,\n        100% {\n            box-shadow: .2em 0px 0 0px currentcolor;\n        }\n        12% {\n            box-shadow: .2em .2em 0 0 currentcolor;\n        }\n        25% {\n            box-shadow: 0 .2em 0 0px currentcolor;\n        }\n        37% {\n            box-shadow: -.2em .2em 0 0 currentcolor;\n        }\n        50% {\n            box-shadow: -.2em 0 0 0 currentcolor;\n        }\n        62% {\n            box-shadow: -.2em -.2em 0 0 currentcolor;\n        }\n        75% {\n            box-shadow: 0px -.2em 0 0 currentcolor;\n        }\n        87% {\n            box-shadow: .2em -.2em 0 0 currentcolor;\n        }\n    }\n"])), function (pr) { return pr.Color ? pr.Color : 'inherit'; });
var mixin = function (size) {
    switch (size) {
        case "XXS":
            return "font-size: 0.8rem;\n            " + Loader + "{\n                margin: 0 2px 2px 2px;\n            }\n           ";
        case "XS":
            return "font-size: 1rem;\n            " + Loader + "{\n                margin: 0 3px 3px 3px;\n            }\n            \n            ";
        case "S":
            return "font-size: 2rem;\n            " + Loader + "{\n                margin: 0 5px 5px 5px;\n            }\n            ";
        case "M":
            return "font-size: 3rem;\n            " + Loader + "{\n                margin: 0 9px 9px 9px;\n            }\n            ";
        case "L":
            return "font-size: 5rem;\n            " + Loader + "{\n                margin: 0 10px 10px 10px;\n            }\n            ";
        case "XL":
            return "font-size: 6rem;\n            " + Loader + "{\n                margin: 0 10px 10px 10px;\n            }\n            ";
        default:
            return "font-size: 1rem;\n        " + Loader + "{\n            margin: 0 3px 3px 3px;\n        }\n       ";
    }
};
export var LoaderWrapper = styled.div(templateObject_2 || (templateObject_2 = __makeTemplateObject(["\n    ", "\n    ", "\n    display: ", ";\n"], ["\n    ", "\n    ", "\n    display: ", ";\n"])), function (pr) { return pr.disabled === true ? 'pointer-events: none;opacity: 0.4;' : ''; }, function (pr) { return mixin(pr.size); }, function (pr) { return pr.display ? 'inline-block' : 'none'; });
var templateObject_1, templateObject_2;
