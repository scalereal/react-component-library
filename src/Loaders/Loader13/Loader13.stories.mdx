import Loader13 from '.';
import { SizeMap } from '../types';
import { Preview, Story, Meta, Props, ArgsTable, Canvas } from '@storybook/addon-docs/blocks';
import { withKnobs, color, boolean, select } from '@storybook/addon-knobs';

<Meta
    title="Components/Loaders/Loader13"
    component={Loader13}
    argTypes={{
        size: {
            control: {
                type: 'select',
                options: Object.values(SizeMap)
            }
        },
        Color: {
            control: 'color'
        },
        display: {
            control: {
                type: 'boolean'
            }
        },
        ref: {
            table: {
                disable: true
            }
        },
        theme: {
            table: {
                disable: true
            }
        },
        as: {
            table: {
                disable: true
            }
        }
    }}
/>

# Loaders

Displays a Loader.

export const Template = args => (
    <div>
        <Loader13 {...args} />
    </div>
);

<Preview>
    <Story name="Loader13" args={{ size: 'S' }}>
        {Template.bind({})}
    </Story>
</Preview>

### Color Variations

You can either change the theme or pass the colors as props.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexWrap: 'wrap' }}>
        <Loader13 size="S" Color="green" />
        <Loader13 size="S" Color="red" />
    </div>
</Preview>

### Sizes

There are Six sizs available `XXS` | `XS` | `S` | `M` | `L` | `XL`.

<Preview withToolbar>
    <div style={{ display: 'flex', justifyContent: 'space-around', flexDirection: 'column' }}>
        <Loader13 size="XXS" />
        <Loader13 size="XS" />
        <Loader13 size="S" />
        <Loader13 size="M" />
        <Loader13 size="L" />
        <Loader13 size="XL" />
    </div>
</Preview>

### Props

<Props story="Loader13" />
