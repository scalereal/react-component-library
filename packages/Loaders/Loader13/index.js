var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from 'react';
import { Loader, LoaderWrapper } from './styled';
var Loader13 = React.memo(React.forwardRef(function (props, ref) {
    var size = props.size;
    return (React.createElement(LoaderWrapper, __assign({ ref: ref, size: size }, props),
        React.createElement(Loader, __assign({}, props))));
}));
Loader13.displayName = "Loader13";
Loader13.defaultProps = {
    size: 'S',
    display: true
};
export default Loader13;
