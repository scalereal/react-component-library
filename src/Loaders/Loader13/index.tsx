import React from 'react';
import { WithStyle } from '../../utils';
import { Props } from '../types';
import { Loader, LoaderWrapper } from './styled';

const Loader13:React.FC<Props> & WithStyle = React.memo(React.forwardRef( (props, ref) => {
    const {size} = props;

    return (
        <LoaderWrapper ref={ref} size={size} {...props}>
            <Loader {...props}/>
        </LoaderWrapper>
    );
    
}));

Loader13.displayName="Loader13";
Loader13.defaultProps = {
    size:'S',
    display: true
};

export default Loader13;
