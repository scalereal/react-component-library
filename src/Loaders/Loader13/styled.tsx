import styled from 'styled-components';
import { Props } from '../types';

export const Loader = styled.div<Props>`
    display: inline-block;
    width: 1em;
    height: 1em;
    color: ${pr=>pr.Color?pr.Color:'inherit'};
    vertical-align: middle;
    pointer-events: none;
    
    transform: rotateZ(45deg);
	perspective: 1000px;
	border-radius: 50%;
	&:before,
	&:after {
		content: '';
		display: block;
		position: absolute;
		top: 0;
		left: 0;
		width: inherit;
		height: inherit;
		border-radius: 50%;
		animation: 1s spin linear infinite;
	}
	&:before {
		transform: rotateX(70deg);
	}
	&:after {
		transform: rotateY(70deg);
		animation-delay: .4s;
	}

    @keyframes rotate {
        0% {
            transform: translate(-50%, -50%) rotateZ(0deg);
        }
        100% {
            transform: translate(-50%, -50%) rotateZ(360deg);
        }
    }

    @keyframes rotateccw {
        0% {
            transform: translate(-50%, -50%) rotate(0deg);
        }
        100% {
            transform: translate(-50%, -50%) rotate(-360deg);
        }
    }

    @keyframes spin {
        0%,
        100% {
            box-shadow: .2em 0px 0 0px currentcolor;
        }
        12% {
            box-shadow: .2em .2em 0 0 currentcolor;
        }
        25% {
            box-shadow: 0 .2em 0 0px currentcolor;
        }
        37% {
            box-shadow: -.2em .2em 0 0 currentcolor;
        }
        50% {
            box-shadow: -.2em 0 0 0 currentcolor;
        }
        62% {
            box-shadow: -.2em -.2em 0 0 currentcolor;
        }
        75% {
            box-shadow: 0px -.2em 0 0 currentcolor;
        }
        87% {
            box-shadow: .2em -.2em 0 0 currentcolor;
        }
    }
`;

const mixin = (size:any) => {
    switch (size) {
        case "XXS":
            return `font-size: 0.8rem;
            ${Loader}{
                margin: 0 2px 2px 2px;
            }
           `;
        case "XS":
            return `font-size: 1rem;
            ${Loader}{
                margin: 0 3px 3px 3px;
            }
            
            `;
        case "S":
            return `font-size: 2rem;
            ${Loader}{
                margin: 0 5px 5px 5px;
            }
            `;
        
        case "M":
            return `font-size: 3rem;
            ${Loader}{
                margin: 0 9px 9px 9px;
            }
            `;

        case "L":
            return `font-size: 5rem;
            ${Loader}{
                margin: 0 10px 10px 10px;
            }
            `;
        
        case "XL":
            return `font-size: 6rem;
            ${Loader}{
                margin: 0 10px 10px 10px;
            }
            `;
  
      default:
        return `font-size: 1rem;
        ${Loader}{
            margin: 0 3px 3px 3px;
        }
       `;
     }
 };

export const LoaderWrapper = styled.div<Props>`
    ${pr => pr.disabled === true?'pointer-events: none;opacity: 0.4;':''}
    ${pr=>  mixin(pr.size)}
    display: ${pr=>pr.visible?'inline-block':'none'};
`;
