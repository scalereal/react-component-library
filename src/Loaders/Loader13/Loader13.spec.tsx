import { render } from '@testing-library/react';
import * as React from 'react';
import Loader13 from '.';

describe('Loader13 Loader', () => {
    it('should render with default theme', () => {
        const { container } = render(<Loader13/>);
        expect(container).toMatchSnapshot();
    });
    it('should render with all the props given', () => {
        const { container } = render(
            <Loader13
                size="M"
                Color='#126AFA'
            />
        );
        expect(container).toMatchSnapshot();
    });

    it('should render image avatar properly', () => {
        const { container } = render(
            <Loader13 size="L" />
        );
        expect(container).toMatchSnapshot();
    });

});