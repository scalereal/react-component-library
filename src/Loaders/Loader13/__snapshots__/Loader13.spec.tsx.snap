// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader13 Loader should render image avatar properly 1`] = `
<div>
  <div
    class="sc-gsTCUz llLtCb"
  >
    <div
      class="sc-bdfBwQ ealFXm"
    />
  </div>
</div>
`;

exports[`Loader13 Loader should render with all the props given 1`] = `
<div>
  <div
    class="sc-gsTCUz liajch"
  >
    <div
      class="sc-bdfBwQ irRoev"
    />
  </div>
</div>
`;

exports[`Loader13 Loader should render with default theme 1`] = `
<div>
  <div
    class="sc-gsTCUz AylWc"
  >
    <div
      class="sc-bdfBwQ ealFXm"
    />
  </div>
</div>
`;
